openstax/pdf-spike
# Trigger PDF Producer (Concourse) Pipeline via Event Service

For Review:
I merged master and then rebased (sorry, still learning), the only files that belong to this PR are:
- `bakery/pipeline-with-service.yml` 
- `bakery/tasks/*`

This PR will allow for the concourse pipeline to be triggered by the workflow-event-resource (new name pending) from cc1.cnx.org, cc2.cnx.org, or cc3.cnx.org.

To Test - 

1.  Update `credentials.yml.template` with the proper credentials. 

2. Navigate to the directory that contains the pipeline configuration file `pipeline-with-service.yml`
```
$ cd /bakery
```

3. Download the Docker Compose file to run Concourse
```
$ wget https://concourse-ci.org/docker-compose.yml
$ docker-compose up -d
Creating docs_concourse-db_1 ...
Creating docs_concourse-db_1 ... done
Creating docs_concourse_1 ...
Creating docs_concourse_1 ... done
Concourse will be running at localhost:8080. You can log in with the username/password as test/test.

Next, install fly CLI by downloading it from the web UI and target your local Concourse as the test user:

$ fly -t pdf-pipeline login -c http://localhost:8080 -u test -p test
logging in to team 'main'

target saved
```

4. Set the Concourse Pipeline
```
fly -t pdf-pipeline set-pipeline -p pdf-producer -c pipeline-with-service.yml -l credentials.yml
```

5. Trigger job by navigating to cc1.cnx.org, cc2.cnx.org, or cc2.cnx.org and "Create a new PDF Job"


To note: 
- The only content server it is currently pulling from is staging ( You can see this hardcoded in on line 20 of `t-fetch-book.yml`), this will be fixed.
